src/parse.ts

```diff
@@ -0,0 +1,263 @@
+        // TODO: configure indent length
+        // forcing name to be defined here, might need a runtime check but it should be set if we are here
+        const descriptionStr = getDescriptionStr(child.name as string, childInfo.description, indentLevel + 1);
+        const optionalStr = child.required ? '' : '?';
+        return `${descriptionStr}\n  ${child.name}${optionalStr}: ${childInfo.tsContent};`;
+      });
+
+      const objectStr = `{\n${childrenContent.join('\n')}\n}`;
+      if (doExport) {
+        return {
+          tsContent: `export interface ${parsedSchema.name} ${objectStr}`,
+          description: parsedSchema.description
+        };
+      }
+      return { tsContent: objectStr, description: parsedSchema.description };
+    }
+    default:
+      throw 'Unsupported join operation ${parsedSchema.joinOperation}';
+  }
+}
+
+export function typeContentToTs(parsedSchema: TypeContent, doExport = false): string {
+  const { tsContent, description } = typeContentToTsHelper(parsedSchema, doExport);
+  // forcing name to be defined here, might need a runtime check but it should be set if we are here
+  const descriptionStr = getDescriptionStr(parsedSchema.name as string, description);
+  return `${descriptionStr}\n${tsContent}`;
+}
+
+// TODO: will be issues with useLabels if a nested schema has a label but is not exported on its own
+
+// TODO: will need to pass around ignoreLabels more
+/**
+ * Parses a joi schema into a TypeContent
+ * @param details: the joi schema
+ * @param Settings: settings used for parsing
+ * @param useLabels if true and if a schema has a label we won't parse it and instead just reference the label in the outputted type
+ * @param ignoreLabels a list a label to ignore if found. Sometimes nested joi schemas will inherit the parents label so we want to ignore that
+ */
+export function parseSchema(
+  details: Describe,
+  settings: Settings,
+  useLabels = true,
+  ignoreLabels: string[] = []
+): TypeContent | undefined {
+  function parseHelper(): TypeContent | undefined {
+    switch (details.type) {
+      case 'array':
+        return parseArray(details, settings);
+      case 'string':
+        return parseStringSchema(details, settings);
+      case 'alternatives':
+        return parseAlternatives(details, settings);
+      case 'object':
+        return parseObjects(details, settings);
+      default:
+        return parseBasicSchema(details, settings);
+    }
+  }
+  const { label, description, required } = getCommonDetails(details, settings);
+  if (label && useLabels && !ignoreLabels.includes(label)) {
+    // skip parsing and just reference the label since we assumed we parsed the schema that the label references
+    // TODO: do we want to use the labels description if we reference it?
+    return makeTypeContentChild({ content: label, customTypes: [label], description, required });
+  }
+  if (!supportedJoiTypes.includes(details.type)) {
+    // TODO: debug/better error logging
```

Comment:
I think an option would be good. It might be good to just make it an `any` type since i think now the property would just be lost if it was in an object. We could even stub out "wrapper" types like `export type Binary = any` so they would at least know what it was in joi.

Down the line if we allow them to pass in custom Ts types we could maybe map the label to their custom type if they know what they want it to parse too

Reply:
`any` works for me

Reply:
Ill create an issue for it